### PR TITLE
Adding dynamixel_interface to documentation index for melodic and noetic

### PR DESCRIPTION
**Pull request by TJMolnar**
# Please Add This Package to be indexed in the rosdistro.

Melodic
Noetic

# The source is here: 

https://github.com/csiro-robotics/dynamixel_interface/tree/noetic-devel

# Checks
 - [x] All packages have a declared license in the package.xml
 - [x] This repository has a LICENSE file
 - [x] This package is expected to build on the submitted rosdistro

### CODE REVIEW

File: melodic/distribution.yaml [hunk 1]
```diff
@@ -2605,6 +2605,16 @@ repositories:
       url: https://github.com/ROBOTIS-GIT/dynamixel-workbench-msgs.git
       version: melodic-devel
     status: developed
+  dynamixel_interface:
+    doc:
+      type: git
+      url: https://github.com/csiro-robotics/dynamixel_interface.git
+      version: noetic-devel
```
Comment (hidmic) [inline melodic/distribution.yaml:2612]: @TJMolnar `noetic-devel` for Melodic?

**Review by hidmic — COMMENTED** [review]
I've noticed this package is requiring C++17 at the CMakeLists.txt level. Bear in mind that any exported includes should be C++14 compliant, as that's the target C++ version for both Melodic and Noetic.

File: melodic/distribution.yaml (hunk 1 above)
Comment (TJMolnar) [reply melodic/distribution.yaml:2612]: @hidmic I have updated the noetic-devel branch to be targeting c++14 explicitly. The branch is specifically compatible with both noetic and melodic so I went with the latest version as a branch name, please advise if a different convention is preferred.

**Comment by TJMolnar** [comment]
@hidmic can you provide some clarity on why nosetests are failing? They are passing on my local checkout of my fork so I'm not sure what needs fixing.

**Comment by wjwwood** [comment]
It kind of looks like this needs to be rebased.

@mergifyio rebase

**Review by mabelzhang — COMMENTED** [review]
Checks are passing.
LICENSE looks okay to me, even though it isn't one of the [OSI Approved Licenses](https://opensource.org/licenses). I don't know enough about licensing to say for sure, so I've asked for internal input.

In the meantime, anyone else more confident about licensing can approve and merge.

**Review by ivanpauno — APPROVED** [review]

**Pull request merged**